File: src/Templates/TypeEditorTemplate/CustomTypeEditor/CustomControl.Design.Server/TypeRoutingProvider.cs
using Microsoft.DotNet.DesignTools.TypeRouting;
using System.Collections.Generic;

namespace CustomControl.Designer.Server
{
    /// <summary>
    /// Class holding the TypeRoutings for resolving the control designer type on the server.
    /// </summary>
    [ExportTypeRoutingDefinitionProvider]
    internal class TypeRoutingProvider : TypeRoutingDefinitionProvider
    {
        public override IEnumerable<TypeRoutingDefinition> GetDefinitions()
            => new[]
            {
                new TypeRoutingDefinition(
                    TypeRoutingKinds.Designer,
                    nameof(CustomControlDesigner),
                    typeof(CustomControlDesigner))
            };
    }
}

File: src/Templates/TemplateSolutions/VB.CustomTypeEditor/CustomControlLibrary.Server/TypeRoutingProvider.vb
Imports Microsoft.DotNet.DesignTools.TypeRouting
Imports System.Collections.Generic

Namespace CustomControlLibrary.Designer.Server
    ''' <summary>
    '''  Class holding the TypeRoutings for resolving the control designer type on the server.
    ''' </summary>
    <ExportTypeRoutingDefinitionProvider>
    Friend Class TypeRoutingProvider
        Inherits TypeRoutingDefinitionProvider

        Public Overrides Function GetDefinitions() As IEnumerable(Of TypeRoutingDefinition)
            Return {New TypeRoutingDefinition(TypeRoutingKinds.Designer, NameOf(CustomControlDesigner), GetType(CustomControlDesigner))}
        End Function
    End Class
End Namespace
